includes/manifest.inc

```diff
@@ -11,94 +11,145 @@
  */
 function islandora_iiif_manifests_create_manifest_json(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_manifest($object);
-
+  
   islandora_iiif_manifests_create_json_response('200 OK', $output);
 }
 
+
 /**
  * Response in JSON for the manifest.
  */
 function islandora_iiif_manifests_create_manifest_datastream(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_manifest($object);
-
+  
   islandora_iiif_manifests_add_manifest_datastream($object, $output);
 }
 
 /**
  * Response in JSON for the canvas.
  */
 function islandora_iiif_manifests_create_canvas_json(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_canvas($object);
-
+  
   islandora_iiif_manifests_create_json_response('200 OK', $output);
 }
 
+/**
+ * Response in JSON for the sub collection
+ */
+function islandora_iiif_manifests_create_collection_json(AbstractObject $object, $collection) {
+  
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  
+  if (!isset($collection)){
+
+    islandora_iiif_manifests_create_json_response('403 Forbidden',"Collection not set" );
+  }else{
+    
+    $output = islandora_iiif_manifests_build_collection($object, $collection);
+  
+    islandora_iiif_manifests_create_json_response('200 OK', $output);
+  }
+ 
+}
+
+
 /**
  * Build manifest array.
  */
 function islandora_iiif_manifests_build_manifest($object) {
-
+  
   $output = array();
-
+  
   $hooks = islandora_build_hook_list(ISLANDORA_IIIF_MANIFESTS_ISLANDORA_IIIF_MANIFEST_HOOK, $object->models);
-
+  
   foreach ($hooks as $hook) {
     $temp = module_invoke_all($hook, $object);
-
+    
     if (!empty($temp)) {
       $output = array_merge_recursive($output, $temp);
     }
   }
-
+  
   return json_encode($output, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE | JSON_NUMERIC_CHECK);
 }
 
 /**
  * Build canvas array.
  */
 function islandora_iiif_manifests_build_canvas($object) {
-
+  
   module_load_include('inc', 'islandora_iiif_manifests', 'includes/utilities');
-
+  
   $output = array();
-
+  
   if (islandora_iiif_manifests_allowed_canvas_cmodel($object)) {
-
+    
     $hooks = islandora_build_hook_list(ISLANDORA_IIIF_MANIFESTS_ISLANDORA_IIIF_MANIFEST_CANVAS_HOOK, $object->models);
-
+    
     foreach ($hooks as $hook) {
       $temp = module_invoke_all($hook, $object);
-
+      
       if (!empty($temp)) {
         $output = array_merge_recursive($output, $temp);
       }
     }
   }
-
+  
   return json_encode($output, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE | JSON_NUMERIC_CHECK);
 }
+
+/**
+ * Build newspaper volume array.
+ */
+function islandora_iiif_manifests_build_collection($object, $collection) {
```

Comment:
Why don't you use the function islandora_iiif_manifests_build_manifest ? It appears to be the same, or isn't it?

Reply:
No, in build collection the collection ID is passed so that the correct collection is returned. 'iiif_manifest/%islandora_object/collection/%

Reply:
The difference is, that in build_collection, the collection ID is being passed so that the correct subcollection is returned when the URL 'iiif_manifest/%islandora_object/collection/%' is called

Reply:
Okay, a difference, but I still think it should only be 1 function: 
rewrite islandora_iiif_manifests_build_manifest that it can get an optional $collection parameter that defaults to NULL

Reply:
You are right. I have updated the code!